docs/guides/mssqlserver/concepts/mssqlserver.md

```diff
@@ -0,0 +1,482 @@
+  namespace: demo
+  resourceVersion: "315403"
+  uid: dafcce02-b6a2-4e65-bdd1-db6b9b6d4913
+type: Opaque
+```
+
+### spec.storageType
+
+`spec.storageType` is an optional field that specifies the type of storage to use for database. It can be either `Durable` or `Ephemeral`. The default value of this field is `Durable`. If `Ephemeral` is used then KubeDB will create MSSQLServer database using [emptyDir](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) volume. In this case, you don't have to specify `spec.storage` field.
+
+### spec.storage
+
+If you don't set `spec.storageType:` to `Ephemeral` then `spec.storage` field is required. This field specifies the StorageClass of PVCs dynamically allocated to store data for the database. This storage spec will be passed to the PetSet created by KubeDB operator to run database pods. You can specify any StorageClass available in your cluster with appropriate resource requests.
+
+- `spec.storage.storageClassName` is the name of the StorageClass used to provision PVCs. PVCs don’t necessarily have to request a class. A PVC with its storageClassName set equal to "" is always interpreted to be requesting a PV with no class, so it can only be bound to PVs with no class (no annotation or one set equal to ""). A PVC with no storageClassName is not quite the same and is treated differently by the cluster depending on whether the DefaultStorageClass admission plugin is turned on.
+- `spec.storage.accessModes` uses the same conventions as Kubernetes PVCs when requesting storage with specific access modes.
+- `spec.storage.resources` can be used to request specific quantities of storage. This follows the same resource model used by PVCs.
+
+To learn how to configure `spec.storage`, please visit the links below:
+
+- https://kubernetes.io/docs/concepts/storage/persistent-volumes/#persistentvolumeclaims
+
+### spec.init
+
+`spec.init` is an optional section that can be used to initialize a newly created MSSQLServer database. MSSQLServer databases can be initialized from these two ways:
+
+1. Initialize from Script
+2. Initialize from Snapshot
+
+### spec.monitor
+
+MSSQLServer managed by KubeDB can be monitored with builtin-Prometheus and Prometheus operator out-of-the-box. To learn more,
+
+- [Monitor MSSQLServer with builtin Prometheus](/docs/guides/mssqlserver/monitoring/using-builtin-prometheus.md)
+- [Monitor MSSQLServer with Prometheus operator](/docs/guides/mssqlserver/monitoring/using-prometheus-operator.md)
+
+### spec.configSecret
+
+`spec.configSecret` is an optional field that allows users to provide custom configuration for MSSQLServer. This field accepts a [`VolumeSource`](https://github.com/kubernetes/api/blob/release-1.11/core/v1/types.go#L47). You can use any Kubernetes supported volume source such as `configMap`, `secret`, `azureDisk` etc. To learn more about how to use a custom configuration file see [here](/docs/guides/mssqlserver/configuration/using-config-file.md).
```

Comment:
Have you tried with `configMap` and `azureDisk` ? I don't think they will work. Check your implementation.
		
```
configSecret, err := r.Client.CoreV1().Secrets(r.db.Namespace).Get(context.TODO(), r.db.Spec.ConfigSecret.Name, meta.GetOptions{})
```

Also the link is broken. Fix the link. 
`[here](/docs/guides/mssqlserver/configuration/using-config-file.md).`